core/tests/behavior/async_write.rs

```diff
@@ -668,20 +684,29 @@ pub async fn test_write_with_append_returns_metadata(op: Operator) -> Result<()>
     let stat_meta = op.stat(&path).await.expect("stat must succeed");
 
     assert_eq!(stat_meta.content_length(), meta.content_length());
-    if cap.write_has_last_modified {
-        assert_eq!(stat_meta.last_modified(), meta.last_modified());
-    }
-    if cap.write_has_etag {
-        assert_eq!(stat_meta.etag(), meta.etag());
-    }
-    if cap.write_has_version {
-        assert_eq!(stat_meta.version(), meta.version());
-    }
-    if cap.write_has_content_md5 {
-        assert_eq!(stat_meta.content_md5(), meta.content_md5());
-    }
-    if cap.write_has_content_type {
-        assert_eq!(stat_meta.content_type(), meta.content_type());
+    if let Some(etag) = meta.etag() {
+        assert_eq!(etag, stat_meta.etag().expect("etag must exist"));
```

Comment:
How about using:

```rust
if meta.etag().is_some() {
    assert_eq!(etag, stat_meta.etag());
}
```

To make sure more easy to read?

Reply:
Looks Good!
How about this?
```
if stat_meta.etag().is_some() && meta.etag().is_some() {
    assert_eq!(stat_etag.etag(), meta.etag());
}

Reply:
This should be treated as a bug if write can return something that stat doesn't have. We either add it in stat or remove it from write returned metadata.

Reply:
Got it.